<!DOCTYPE html>
<html lang="en">
<html class="dark light">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    

    
    
    
    <title>
         American Society of Human Genetics meeting 2025
        
    </title>

        
            <meta property="og:title" content="American Society of Human Genetics meeting 2025" />
        
     

     
         
     

     
         
    

    
    
        <link rel="icon" type="image/png" href=&#x2F;icon&#x2F;favicon.ico />
    

    
    
        <link href=https://ypark.github.io/fonts.css rel="stylesheet" />
    

    
    

    
    
        <script src=https://ypark.github.io/js/codeblock.js></script>
    

    
    
        <script src=https://ypark.github.io/js/toc.js></script>
    
    
    
    

    

	<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/MathJax.js?config=TeX-MML-AM_CHTML"></script>

        <script type="text/javascript" id="MathJax-script" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js">
        </script>
        
            <script>
            MathJax = {
              tex: {
                  inlineMath: [['$', '$'], ['\\(', '\\)'], ['$$', '$$']]
              }
            };
            </script>
        

	<!-- <\!-- KaTeX CSS -\-> -->
	<!-- <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.15.2/katex.min.css"> -->
	
	<!-- <\!-- KaTeX JS -\-> -->
	<!-- <script src="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.15.2/katex.min.js"></script> -->
	
	<!-- <\!-- Optional: KaTeX Auto-render extension -\-> -->
	<!-- <script src="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.15.2/contrib/auto-render.min.js"></script> -->
	
	<!-- <script> -->
	<!--   document.addEventListener("DOMContentLoaded", function() { -->
	<!--       renderMathInElement(document.body, { -->
	<!-- 	  delimiters: [ -->
	<!-- 	      {left: "$$", right: "$$", display: true}, -->
	<!-- 	      {left: "$", right: "$", display: false} -->
	<!-- 	  ] -->
	<!--       }); -->
	<!--   }); -->
	<!-- </script> -->

    

    
    <link rel="alternate" type="application/atom+xml" title="" href="https://ypark.github.io/atom.xml">


    
    
        <link rel="stylesheet" type="text/css" href=https://ypark.github.io/theme/light.css />
        <link id="darkModeStyle" rel="stylesheet" type="text/css" href="https://ypark.github.io/theme/dark.css" />
    

    <!-- Set the correct theme in the script -->
    <script src=https://ypark.github.io/js/themetoggle.js></script>
    
        <script>setTheme(getSavedTheme());</script>
    

    <link rel="stylesheet" type="text/css" media="screen" href=https://ypark.github.io/main.css />

    
</head>


<body>
    <div class="content">
        <header>
    <div class="main">
        <a href=https:&#x2F;&#x2F;ypark.github.io></a>

        <div class="socials">
            
            <a rel="me" href="https:&#x2F;&#x2F;github.com&#x2F;causalpathlab&#x2F;" class="social">
                <img alt=github src=https://ypark.github.io/social_icons/github.svg>
            </a>
            
        </div>
    </div>

    <nav>
        
        <a href=https://ypark.github.io style="margin-left: 0.5em">home</a>
        
        <a href=https://ypark.github.io/about style="margin-left: 0.5em">&#x2F;about</a>
        
        <a href=https://ypark.github.io/posts style="margin-left: 0.5em">&#x2F;posts</a>
        
        <a href=https://ypark.github.io/team style="margin-left: 0.5em">&#x2F;team</a>
        
        <a href=https://ypark.github.io/alumni style="margin-left: 0.5em">&#x2F;alumni</a>
        

        
        |<a id="dark-mode-toggle" onclick="toggleTheme(); event.preventDefault();" href="#">
            <img src=https://ypark.github.io/feather/sun.svg id="sun-icon" style="filter: invert(1);" alt="Light" />
            <img src=https://ypark.github.io/feather/moon.svg id="moon-icon" alt="Dark" />
        </a>

        <!-- Inititialize the theme toggle icons -->
        <script>updateItemToggleTheme()</script>
        
    </nav>
</header>


        
        
    
<main>
  <article>
    <div class="title">
      
      
<div class="page-header">
  American Society of Human Genetics meeting 2025<span class="primary-color" style="font-size: 1.6em">.</span>
</div>


      <div class="meta">
        
        Posted on <time>2025-10-16</time>
        

        

        

        
        
        <span class="tags-label"> :: Tags:</span>
        <span class="tags">
          <a href="https://ypark.github.io/tags/conference/" class="post-tag">conference</a>, 
          
          <a href="https://ypark.github.io/tags/genetics/" class="post-tag">genetics</a>, 
          
          <a href="https://ypark.github.io/tags/thoughts/" class="post-tag">thoughts</a>, 
          
          <a href="https://ypark.github.io/tags/spatial-omics/" class="post-tag">spatial omics</a>, 
          
          <a href="https://ypark.github.io/tags/idea-flow/" class="post-tag">idea flow</a>
          
        </span>
        

        
        
        
        
        
        

        
        

        

      </div>
    </div>

    

    
    
    
    <div class="toc-container">
      <h1 class="toc-title">Table of Contents</h1>
      <ul class="toc-list">
        
        <li>
          <a href="https://ypark.github.io/posts/ashg/#oct-15">Oct 15</a>
          
          <ul>
            
            <li>
              <a href="https://ypark.github.io/posts/ashg/#non-linear-gwas-methods">Non-linear GWAS methods</a>
            </li>

            
            
            <li>
              <a href="https://ypark.github.io/posts/ashg/#context-specific-polygenic-score-models">Context-specific polygenic score models</a>
            </li>

            
            
            <li>
              <a href="https://ypark.github.io/posts/ashg/#how-can-include-gene-gene-interactions-in-mr">How can include gene-gene interactions in MR?</a>
            </li>

            
            
          </ul>
          
        </li>
        
        <li>
          <a href="https://ypark.github.io/posts/ashg/#oct-16">Oct 16</a>
          
          <ul>
            
            <li>
              <a href="https://ypark.github.io/posts/ashg/#distinguishing-causal-variants-underlying-natural-selection">Distinguishing causal variants underlying natural selection</a>
            </li>

            
            
            <li>
              <a href="https://ypark.github.io/posts/ashg/#pubmind-language-model-trained-on-pubmed">PubMind language model trained on PUBMED</a>
            </li>

            
            
            <li>
              <a href="https://ypark.github.io/posts/ashg/#spatial-3d-reconstruction-of-h-e-data">Spatial 3D reconstruction of H&amp;E data</a>
            </li>

            
            
            <li>
              <a href="https://ypark.github.io/posts/ashg/#xenotransplantation">Xenotransplantation</a>
            </li>

            
            
            <li>
              <a href="https://ypark.github.io/posts/ashg/#mtdna-based-linage-tracing-in-spatial-data">MtDNA-based linage tracing in spatial data</a>
            </li>

            
            
            <li>
              <a href="https://ypark.github.io/posts/ashg/#spatial-neighbour-analysis">Spatial neighbour analysis</a>
            </li>

            
            
            <li>
              <a href="https://ypark.github.io/posts/ashg/#clonal-hematopoiesis-non-coding-putative-driver-mutations">Clonal hematopoiesis, non-coding putative driver mutations</a>
            </li>

            
            
          </ul>
          
        </li>
        
        <li>
          <a href="https://ypark.github.io/posts/ashg/#after-the-ashg-meeting">After the ASHG meeting</a>
          
        </li>
        
      </ul>
    </div>
    
    

    <section class="body">
      <p>TL;DR: Biased memos from ASHG 2025 talks/posters. AI/ML and new omics technologies always create new interests in the genetics community, but certain aspects of core genetic questions aren't easily changed. Large-scale biobank GWAS data became the norm; then what's next? Can we think of anything beyond polygenic risk predictions? Contextual information matters, but how so?</p>
<h2 id="oct-15"><a class="zola-anchor" href="#oct-15" aria-label="Anchor link for: oct-15">Oct 15</a></h2>
<h3 id="non-linear-gwas-methods"><a class="zola-anchor" href="#non-linear-gwas-methods" aria-label="Anchor link for: non-linear-gwas-methods">Non-linear GWAS methods</a></h3>
<ul>
<li>Refine additive association models with dominant and recessive models</li>
<li>Concerns over Type I error calibration</li>
<li>How is it different from model checking?</li>
</ul>
<h3 id="context-specific-polygenic-score-models"><a class="zola-anchor" href="#context-specific-polygenic-score-models" aria-label="Anchor link for: context-specific-polygenic-score-models">Context-specific polygenic score models</a></h3>
<ul>
<li>Fonseca, Andrew Dahl, U of Chicago</li>
<li>Improve prediction performance
<ul>
<li>Can contexts be anything?</li>
</ul>
</li>
<li>Straightforward idea: calculate interaction GWAS</li>
<li>Build PGS models base off interaction coefficients</li>
<li>Can we come up with summary-based extensions?</li>
</ul>
<h3 id="how-can-include-gene-gene-interactions-in-mr"><a class="zola-anchor" href="#how-can-include-gene-gene-interactions-in-mr" aria-label="Anchor link for: how-can-include-gene-gene-interactions-in-mr">How can include gene-gene interactions in MR?</a></h3>
<ul>
<li>No satisfactory answers</li>
</ul>
<h2 id="oct-16"><a class="zola-anchor" href="#oct-16" aria-label="Anchor link for: oct-16">Oct 16</a></h2>
<ul>
<li>Squeezing AI models into genetic</li>
</ul>
<h3 id="distinguishing-causal-variants-underlying-natural-selection"><a class="zola-anchor" href="#distinguishing-causal-variants-underlying-natural-selection" aria-label="Anchor link for: distinguishing-causal-variants-underlying-natural-selection">Distinguishing causal variants underlying natural selection</a></h3>
<ul>
<li>
<p>Arya Rao, Sabeti Lab (Harvard), Steven Riley (Yale), DeepSweep</p>
</li>
<li>
<p>Selective sweep: SLC45A2, causal adaptive variants</p>
</li>
<li>
<p>Distinguish causal and neural-hitchhiker variants</p>
</li>
<li>
<p>Simulated data (coalescence model). Summary statistics.</p>
</li>
<li>
<p>Transformer-based prediction (do we need label data?)</p>
</li>
<li>
<p>GWAS and eQTL enrichment</p>
</li>
<li>
<p>But where is causal guarantee?</p>
</li>
<li>
<p>Question from audience:</p>
<ul>
<li>Have you compared with colocalized or finemapping results?</li>
<li>Do ages of variants matter?</li>
</ul>
</li>
<li>
<p>Takeaway: Probably useful... run simulation and train black box models.</p>
</li>
</ul>
<h3 id="pubmind-language-model-trained-on-pubmed"><a class="zola-anchor" href="#pubmind-language-model-trained-on-pubmed" aria-label="Anchor link for: pubmind-language-model-trained-on-pubmed"><code>PubMind</code> language model trained on PUBMED</a></h3>
<ul>
<li>Train LLM on millions of abstracts</li>
<li>Existing resources: ClinVar, HGMD, LitVar2, ...</li>
<li>inference module: SNV, SV, CNV, Gene fusion</li>
<li>Focus on genetic evidence of pathogenecity</li>
<li>What are the queries?</li>
<li>LLaMa3.3, DeepSeek</li>
<li>Standardize disease name</li>
<li><a href="https://pubmind.wglab.org"><code>https://pubmind.wglab.org</code></a></li>
<li>This is still a database... but the benefit of LLM is interactive API</li>
<li>We could find more, but how should we use the method?</li>
</ul>
<h3 id="spatial-3d-reconstruction-of-h-e-data"><a class="zola-anchor" href="#spatial-3d-reconstruction-of-h-e-data" aria-label="Anchor link for: spatial-3d-reconstruction-of-h-e-data">Spatial 3D reconstruction of H&amp;E data</a></h3>
<ul>
<li>Hu, Yu, Emory</li>
<li>Use long range dependency between snapshots</li>
<li>Multi-resolution alignment</li>
<li>Metamorphism (gradient flow)</li>
<li>Cool data: CyCIF + H&amp;E on colorectal cancer</li>
<li>Cool 3D image... What did we learn?
<ul>
<li>Can we not find the same cell type identity mapping?</li>
</ul>
</li>
<li>A question from moderator: Any downstream analysis?</li>
</ul>
<h3 id="xenotransplantation"><a class="zola-anchor" href="#xenotransplantation" aria-label="Anchor link for: xenotransplantation">Xenotransplantation</a></h3>
<ul>
<li>Eloi Schmauch, Keating, NYU</li>
<li>Pig kidney transplantation: pig vs. human cells?</li>
<li>Too much technology details? Clustering?</li>
<li>Visualization by <a href="https://github.com/broadinstitute/celldega"><code>Celldega</code></a></li>
<li>Cell type colocalization over the course of transplantation</li>
<li>Questions: How good is the pig genome?</li>
<li>Suggestions: Try out different technologies for adjacent slides?</li>
</ul>
<h3 id="mtdna-based-linage-tracing-in-spatial-data"><a class="zola-anchor" href="#mtdna-based-linage-tracing-in-spatial-data" aria-label="Anchor link for: mtdna-based-linage-tracing-in-spatial-data">MtDNA-based linage tracing in spatial data</a></h3>
<ul>
<li>Rong, Zhang, UPenn</li>
<li>mtDNA far more abundant in cells</li>
<li>Can linage information inferred from mtDNA implicate spatial changes?</li>
<li>Send cDNA to (1) spatial slide-seq (2) PCR amplification for genotyping</li>
<li>Input: MT Loci with cell type deconvolution (spots)</li>
<li>Question: Which variant is significantly over-representing cell types?</li>
<li>Assign MT variants to cell types</li>
<li>Barrett's Esophagus</li>
<li>Bracht and Rong <em>et al.</em> (2025)</li>
<li>Spot-level localization (testing the purity of nearest neighbors)</li>
</ul>
<p>Can we simply use MT expression to construct cell lineage?
Why not using expression-based CNV method?</p>
<h3 id="spatial-neighbour-analysis"><a class="zola-anchor" href="#spatial-neighbour-analysis" aria-label="Anchor link for: spatial-neighbour-analysis">Spatial neighbour analysis</a></h3>
<ul>
<li>Jun Inamo, U of Colorado</li>
<li>Juvenile Idiopathic Arthritis synovium</li>
<li>10x Xenium, n=9</li>
<li>co-varying neighborhood analysis (<a href="https://www.nature.com/articles/s41587-021-01066-4">Reshef YA, Nat Biotech 2022</a>)</li>
<li>Can we relate this with locally-linear embedding?</li>
<li>Show dynamics of many fibroblast marker genes as a function of distance from endothelial cells</li>
<li><code>spatialCoocur</code> (by permutation tests)</li>
<li>Compute colocalization scores between two cell types (e.g., T and myeloid, <em>CXCL9</em>-<em>CXCR3</em>)</li>
<li>Tertiary Lymphoid Structure (TLS): T-cell and B-cell together</li>
</ul>
<h3 id="clonal-hematopoiesis-non-coding-putative-driver-mutations"><a class="zola-anchor" href="#clonal-hematopoiesis-non-coding-putative-driver-mutations" aria-label="Anchor link for: clonal-hematopoiesis-non-coding-putative-driver-mutations">Clonal hematopoiesis, non-coding putative driver mutations</a></h3>
<ul>
<li>Josh Weinstock, Emory</li>
<li>How does DNA changes over time?</li>
<li>Clonal hematopoiesis of indeterminate potential</li>
<li>Clonal extensions lack a mutation in canonical genes.</li>
<li>490k UKBB whole genome</li>
<li>frequency of alternate allele as a function of age at blood draw</li>
<li>clonal hematopoiesis genes recapitulated!</li>
<li>AML risk was also captured in phewas on the variants</li>
<li>CH point mutations... how much variance was explained?</li>
<li><a href="https://somatic.emory.edu">somatic.emory.edu</a></li>
<li><a href="https://www.medrxiv.org/content/10.1101/2025.10.12.25337792v1">preprint</a></li>
<li>Questions from audience:
<ul>
<li>How do we know this is somatic or rare germline?</li>
<li>Most of them are germline point mutations.</li>
<li>Do cell types matter?</li>
<li>Copy number variants?</li>
</ul>
</li>
<li>We may be able to ascertain "age" as a factor for somatic vs. germline... <em>Does it matter?</em></li>
</ul>
<h2 id="after-the-ashg-meeting"><a class="zola-anchor" href="#after-the-ashg-meeting" aria-label="Anchor link for: after-the-ashg-meeting">After the ASHG meeting</a></h2>

    </section>
  </article>
</main>



        

    </div>
</body>

</html>
